build/docs/troubleshooting.md

```diff
@@ -0,0 +1,70 @@
+# Troubleshooting Guide
+
+This guide covers frequent issues and possible solutions when developing with Agones.
+
+## Common Development Issues
+
+### $GOPATH/$GOROOT error when building in WSL
+
+If you get this error when building Agones in WSL (`make build`, `make test` or any other related target):
+
+```
+can't load package: package agones.dev/agones/cmd/controller: cannot find package "agones.dev/agones/cmd/controller" in any of:
+       /usr/local/go/src/agones.dev/agones/cmd/controller (from $GOROOT)
+       /go/src/agones.dev/agones/cmd/controller (from $GOPATH)
+```
+
+- Are your project files on a different folder than C? If yes, then you should either move them on drive C or set up Docker for Windows to share your project drive as well
+- Did you set up the volume mount for Docker correctly? By default, drive C is mapped by WSL as /mnt/c, but Docker expects it as /c. You can test by executing `ls /c` in your linux shell. If you get an error, then follow the instructions for [setting up volume mount for Docker](https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly#ensure-volume-mounts-work)
+
+### Error: cluster-admin-binding already exists
+
+This surfaces while running `make gcloud-auth-cluster`. The solution is to run `kubectl describe clusterrolebinding | grep cluster-admin-binding- -A10`, find clusterrolebinding which belongs to your `User` account and then run `kubectl delete clusterrolebindings cluster-admin-binding-<md5Hash>` where `<md5Hash>` is a value specific to your account. Now you can execute `make gcloud-auth-cluster` again. If you run into a permission denied error when attempting the delete operation, you need to run `sudo chown <your username> <path to .kube/config>` to change ownership of the file to yourself.
+
+### Error: releases do not exist
+
+Run `make uninstall` then run `make install` again.
+
+## Performance and Profiling Issues
+
+### I want to use pprof to profile the controller.
```

Comment:
Should we move this pprof guide to development-workflow.md? Seems like it would be a good fit?